### PR TITLE
Make dedicated pill shape function

### PR DESCRIPTION
**Pull request by ZeLonewolf**
This PR adds "pill" as a first-class shape that a shield style can specify.  Behind the scenes in the shield library, it still just invoking roundedRectangle, but this allows the shield style definition to not have to do the hack of specifying a pill by declaring a rounded rectangle with a large radius. This removes the hack mentioned in #1062 for taginfo and saves a few bytes of ShieldJSON.

I anticipate doing the same job for circle and homePlate shapes.

### CODE REVIEW

File: shieldlib/src/shield_canvas_draw.ts [hunk 1]
```diff
@@ -97,6 +97,16 @@ function ellipse(
   return width;
 }
 
+export function pill(
+  r: ShieldRenderingContext,
+  ctx: CanvasRenderingContext2D,
+  params: ShapeBlankParams,
+  ref?: string
+): number {
+  params.radius = r.shieldSize() / 2;
```
Comment (claysmalley) [inline shieldlib/src/shield_canvas_draw.ts:106]: This doesn't seem to be working right for 200% zoom:

[image](https://preview.ourmap.us/pr/1069/#map=14.2/38.19356/-81.49928)

**Review by claysmalley — COMMENTED** [review]
> I anticipate doing the same job for circle and homePlate shapes.

How come? Because they have different `prettyShapeName`s?

**Comment by ZeLonewolf** [comment]
> How come? Because they have different `prettyShapeName`s?

I think it's generally useful and more readable in the ShieldJSON to have the shield definition be able to specify common shapes by name, particularly when then can consume defaults (in this case, corner radii). I confess that I don't fully understand @1ec5's vision for shield definition re-usability, but I see value in a slightly richer dictionary of shield shapes being more extendable down the line.

File: shieldlib/src/shield_canvas_draw.ts (hunk 1 above)
Comment (ZeLonewolf) [reply shieldlib/src/shield_canvas_draw.ts]: Fixed in dd2192e. I also added a note explaining the logic at that spot in the code.

**Review by claysmalley — APPROVED** [review]

**Pull request merged**